plugins/social_network/social_network.cpp

```diff
@@ -263,7 +259,16 @@ namespace golos { namespace plugins { namespace social_network {
 
     void social_network::impl::set_url(discussion& d) const {
         const comment_api_object root(database().get<comment_object, by_id>(d.root_comment));
-        d.url = std::string("/") + root.category + "/@" + root.author + "/" + root.permlink;
+
+        tags::comment_metadata meta = tags::operation_visitor::get_metadata(root);
+
+        if( root.json_metadata.size() && !meta.tags.empty()) {
```

Comment:
`root.json.metadata.size()` is a superfluous condition.
Here should be: 
```
if (!meta.tags.empty()) {
    d.url = "/" + *meta.tags.begin() + "/@" + root.author + "/" + root.permlink;
} else {
    d.url = "/@" + root.author + "/" + root.permlink;
}
```

Reply:
Ok il fix